mobile/lib/modules/asset_viewer/views/gallery_viewer.dart

```diff
@@ -186,7 +186,9 @@ class GalleryViewerPage extends HookConsumerWidget {
         } else {
           SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersive);
         }
-        isPlayingVideo.value = false;
+        var autoPlayMotionPhotos =
+            settings.getSetting<bool>(AppSettingsEnum.autoPlayVideos);
+        isPlayingVideo.value = isMotionPhoto && autoPlayMotionPhotos;
```

Comment:
Do not modify the `isPlayingVideo` directly. This is used to rebuild the gallery viewer when a video is playing. Instead, have a separate state called `shouldAutoPlay` similar to `isLoadOriginal`, assign the value during `useEffect` and use it as the argument to the `VideoViewerPage`

Reply:
If not like this, I'm unsure how to make motion pictures autoplay, further down that is exactly how the "Play Motion Picture" Button in the `GalleryAppBar` works:

`onToggleMotionVideo: () =>
                    isPlayingVideo.value = !isPlayingVideo.value`